nixos/tests/systemd-sysusers-mutable.nix

```diff
@@ -0,0 +1,71 @@
+{ lib, ... }:
+
+let
+  rootPassword = "$y$j9T$p6OI0WN7.rSfZBOijjRdR.$xUOA2MTcB48ac.9Oc5fz8cxwLv1mMqabnn333iOzSA6";
+  normaloPassword = "hello";
+  newNormaloPassword = "$y$j9T$p6OI0WN7.rSfZBOijjRdR.$xUOA2MTcB48ac.9Oc5fz8cxwLv1mMqabnn333iOzSA6";
+in
+
+{
+
+  name = "activation-sysusers-mutable";
+
+  meta.maintainers = with lib.maintainers; [ nikstur ];
+
+  nodes.machine = { pkgs, ... }: {
+    systemd.sysusers.enable = true;
+    users.mutableUsers = true;
+
+    # Prerequisites
+    system.etc.overlay.enable = true;
+    boot.initrd.systemd.enable = true;
+    boot.kernelPackages = pkgs.linuxPackages_latest;
+
+    # Override the empty root password set by the test instrumentation
+    users.users.root.hashedPasswordFile = lib.mkForce null;
+    users.users.root.initialHashedPassword = rootPassword;
+    users.users.normalo = {
+      isNormalUser = true;
```

Comment:
maybe this was anyway mentioned but what about [this](https://manpages.ubuntu.com/manpages/bionic/man5/sysusers.d.5.html)?

> systemd-sysusers uses the files from sysusers.d directory to create system users and
       groups at package installation or boot time. This tool may be used to allocate system
       users and groups only, it is not useful for creating non-system (i.e. regular, "human")
       users and groups, as it accesses [/etc/passwd](file:///etc/passwd) and [/etc/group](file:///etc/group) directly, bypassing any more
       complex user databases, for example any database involving NIS or LDAP.

Reply:
Do you mean their differentiation between system users and normal users? In the systemd universe "normal" users are only those managed by homed. "system" users are those that get an entry in /etc/passwd and friends. So I'd say NixOS only ever had "system" users from the systemd perspective.

Reply:
ah thanks

for some reason in my mind i thought this was tied to id ranges... must have got my wires crossed